### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by coderschoolreview** [dotruongan/HomeworkSubmission#1]
![](http://i.imgur.com/psFPti3.png) 
Two error in this code:
- If you want to access a column in pandas data frame, use square bracket notation. 
Eg:
 ```python 
selected_rows = ecom['CC Exp Date']
```

- CC exp Date is a string, formated by `mm/yy`. Can't make comparison between a string and a integer `selected_rows = ecom('CC Exp Date') == 2025`

For data visualization, please read :
- [Seaborn barplot](https://seaborn.pydata.org/generated/seaborn.barplot.html)
-[Seaborn distribution plot](https://seaborn.pydata.org/tutorial/distributions.html)

**Comment by coderschoolreview** [comment]
![](http://i.imgur.com/68OzaoM.png) 

```python
sns.distplot(x=data('MAE'),y=('MSE'),data=data)
```

It's good you can lookup on google for yourself, it's good way to learn

**Comment by coderschoolreview** [comment]
Didn't complete training model.
Can take a look in this notebook to complete https://github.com/huynhtan126/Homework_03/blob/master/Assignment_3-work.ipynb

**Comment by coderschoolreview** [comment]
#Assigment 4:
## EDA:
* Has some useful EDA, like the scatter plot
![](http://i.imgur.com/3vSQ8Gg.png)

* Should try some other eda, using `sns.pairplot`, `sns.distplot`. Eg: 
```python
# Distribution of length between 3 species: 
## list of species
ls = list(data.Species.unique())

plt.rcParams['figure.figsize'] = (20,5)
f,axes = plt.subplots(1,4)

i=0;
for v in ('SepalLengthCm','SepalWidthCm','PetalLengthCm','PetalWidthCm'):
    plt.sca(axes[i])
    axes[i].set_title(v)
    
    for s in ls:
        sns.distplot(data[data.Species==s][v],
                     hist=False,
                     label=s
                    )
    i+=1
    
plt.legend()
plt.show()

```
![](http://i.imgur.com/HsMJwSd.png)
* For ref, check out [Mai's notebook](https://github.com/janetvn/coderschool_mc_w04_assignment))


## Model: 
* You're not really understand what you're doing, do you?

**Comment by coderschoolreview** [comment]
# An Do: Assignment 5

The goal of this assignment was to introduce you following concepts in Machine Learning:

- Data quality check: missing value, anomaly, ...
- Exploratory Data Analysis: ratio of class, relationship of feature with another and relationship of feature with target, and data visualization.
- Preprocessing data: handling imbalance dataset, dealing with categorical variable.
- Modeling & evaluation: building classifier and evaluate, Hyper Parameter Tuning.

## Things you did well

* Checking data quality
* EDA: Ratio of classes, finding out that data is imbalance, visualizing relationship between features 
* Preprocessing: train test split.

## Things to work on

* Preprocessing categorical features. Try `pandas.get_dummies`
* Preprocessing under sampling / over sampling
* Modeling & evaluate: train and evaluate multiple model by comparing classification report of models against each other
* GridSearchCV and RandomSearchCV: common practice is using RandomSearchCV to narrow down search range then fine tuning with GridSearchCV, can read [this article](https://towardsdatascience.com/hyperparameter-tuning-the-random-forest-in-python-using-scikit-learn-28d2aa77dd74)

## Minor tips

* Installing packages `missingno ` (or any arbitrary package) on Win10:
  * Go to `Anaconda Prompt`
  * Type `conda install -c conda-forge missingno`

* Filtering series: 

  * Your code:`train_copy.isnull().sum()`

  * Could write like this to get column with null value only:

    ```python
    ncols = train_copy.isnull().sum
    ncols[ncols!=0]
    ```

* For evaluation function, should `print` instead of `return` so when you loop through list of model and evaluate them, the result for each iteration printed to output.

**Comment by coderschoolreview** [comment]
---
# Assignment 6

# Goal of this Assignment

The goal of this assignment was to introduce you to following concepts:
* Unsupervised Learning (KMeans, Hierachical Clustering)
* PCA

You learn how to use PCA for dimension reduction, KMeans, and Hierarchical Clustering. Also you learn to visualize the result of both tenichque.

# Things you did well:

* Ultilizing Pandas to read, summary and visualize data.
* Got everything right


# To sum up:

* Great work! This showing you understood the concept and able to apply it. Keep it up!

# Minor tips:
![](http://i.imgur.com/DJlcIId.png)

This is not what the question asking for, try this

```python
total_purchases = data.sum(axis=1)
purchase_percent = data.div(total_purchases, axis=0) * 100
```

![](http://i.imgur.com/saIkTR3.png)

Goal of this question is to sort users by total amount spending across all category. (`.sum(axis=1)` sum every row across columns and return total spending for each user.
Should be `data.sum(axis=1).nsmallest(44)`
Or 
```python
select_lowest = data.sum(axis=1) < data.sum(axis=1).quantile(.1)
data[select_lowest].index
```
